### PR TITLE
UIOR-1256: Add additional order filters to support reporting requirements

### PR DESCRIPTION
**Pull request by alisher-epam**
## Purpose

https://folio-org.atlassian.net/browse/UIOR-1256 - UIOR-1256: Add additional order filters to support reporting requirements
## Approach

https://github.com/folio-org/ui-orders/assets/116072773/2916ffb4-24bc-445c-b522-3db711c722a2

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/common/ExportSettingsModal/utils/createExportReport.js [hunk 1]
```diff
@@ -132,6 +136,7 @@ const getOrderLineExportData = ({
   locationMap,
   materialTypeMap,
   poLinesMap,
+  userMap,
```
Comment (NikitaSedyx) [inline src/common/ExportSettingsModal/utils/createExportReport.js:139]: I don't think it includes info from order line, look at my fixes for invoice and receiving

File: src/common/ExportSettingsModal/utils/createExportReport.js [hunk 2]
```diff
@@ -118,6 +118,10 @@ const getOrderExportData = ({
     renewalDate: formatDate(order.ongoing?.renewalDate, intl),
     reviewDate: formatDate(order.ongoing?.reviewDate, intl),
     poTags: order.tags?.tagList?.join('|'),
+    createdBy: userMap[order.metadata?.createdByUserId]?.username ?? invalidReference,
```
[the remark this answers is not in the document]
Comment (alisher-epam) [reply src/common/ExportSettingsModal/utils/createExportReport.js:121]: It includes it. [git](https://github.com/folio-org/ui-orders/blob/master/src/common/ExportSettingsModal/utils/getExportData.js#L25-L27)

File: src/common/ExportSettingsModal/utils/createExportReport.js (hunk 1 above)
Comment (alisher-epam) [reply src/common/ExportSettingsModal/utils/createExportReport.js:139]: I am extending it

**Comment by NikitaSedyx** [comment]
plz include required permissions

**Review by NikitaSedyx — APPROVED** [review]

**Review by usavkov-epam — APPROVED** [review]

**Pull request merged**